packages/aws-cdk-lib/aws-stepfunctions/README.md

```diff
@@ -467,8 +467,27 @@ execute the same steps for multiple entries of an array in the state input.
 const map = new sfn.Map(this, 'Map State', {
   maxConcurrency: 1,
   itemsPath: sfn.JsonPath.stringAt('$.inputForMap'),
+  parameters: {
+    item: sfn.JsonPath.stringAt('$$.Map.Item.Value'),
+  },
+  resultPath: '$.mapOutput',
 });
-map.iterator(new sfn.Pass(this, 'Pass State'));
+
+// The Map iterator can contain a IChainable, which can be an individual or multiple steps chained together.
+// Below example is with a Choice and Pass step
+const choice = new sfn.Choice(this, 'Choice');
+const condition1 = sfn.Condition.stringEquals('$.item.status', 'SUCCESS');
+const step1 = new sfn.Pass(this, 'Step1');
+const step2 = new sfn.Pass(this, 'Step2');
+const finish = new sfn.Pass(this, 'Finish');
+
+const definition = choice
+    .when(condition1, step1)
+    .otherwise(step2)
+    .afterwards())
```

Comment:
```suggestion
    .afterwards()
```

Reply:
Thanks so much @kaizencc for feedback.
I am running into  `JS heap out of memory` issue when I run  `yarn rosetta:extract --strict` (hence having to publish to this draft PR after I make changes and note the build failures). 

Are there any known workarounds? I tried increasing mem size using `NODE_OPTIONS="--max-old-space-size=819` but to no avail

Reply:
Also posted my query on cdk.dev slack for ref https://cdk-dev.slack.com/archives/C041656G340/p1692825834382109

Reply:
I am not sure, but it is definitely a memory intensive process that may take more memory than `yarn build`. Can you create an issue for it? That's probably the best thing to do here. Thanks for your contribution anyway!

Reply:
Thanks @kaizencc for approving, will do!